system/jlib/jsecrets.cpp

```diff
@@ -1195,28 +1170,33 @@ class CSecret final : public CInterfaceOf<ISecret>
         updateHash();
     }
 
-    virtual const IPropertyTree * getTree() const;
+    virtual const IPropertyTree * getTree() const override;
 
-    virtual bool getKeyValue(MemoryBuffer & result, const char * key) const
+    virtual bool getProp(MemoryBuffer & result, const char * key) const override
     {
         CriticalBlock block(secretCs);
         checkStale();
         return getSecretKeyValue(result, secret, key);
     }
-    virtual bool getKeyValue(StringBuffer & result, const char * key) const
+    virtual bool getProp(StringBuffer & result, const char * key) const override
     {
         CriticalBlock block(secretCs);
         checkStale();
         return getSecretKeyValue(result, secret, key);
     }
-    virtual bool isStale() const
+    virtual bool isStale() const override
     {
         return secret && hasCacheExpired(secret);
     }
-    virtual unsigned getVersion() const
+    virtual unsigned getVersion() const override
     {
         return secretHash;
     }
+    virtual bool isValid() const override
+    {
+        CriticalBlock block(secretCs);
```

Comment:
What is the point of an isValid() function like this? As soon as you release the critsec, the value you are returning may no longer be correct...

Reply:
This is typically called at startup to check that the configuration has been set up.  The calling code is cleaner since it avoids the problem with freeing the returning the linked object.